MRG: _TimeViewer resource guide (#7606)
mne-tools/mne-python: mne/images/README.rst

.. -*- mode: rst -*-

Documentation
=============

> The images are used in `mne/viz/_brain` as icons in the toolbar of `_TimeViewer`.

Reviewer: reStructuredText needs double backticks (2 on each end) to trigger code formatting.  Currently "mne/viz/_brain" and "_TimeViewer" will be in italics. Relevant throughout this file.

> It is necessary to build those images into a resource file for proper usage by

Reviewer 2: ```suggestion
It is necessary to compile those images into a resource file for proper use by
```

the application.

The resource configuration file `mne.qrc` describes the location of the resources
in the filesystem and also defines aliases for their use in the code.

> To generate automatically the resource file in `mne/viz/_brain`:

Reviewer 2: ```suggestion
To automatically generate the resource file in `mne/viz/_brain`:
```

.. code-block:: bash

    pyrcc5 -o mne/viz/_brain/resources.py mne.qrc

Patching
========

> The output file imports `PyQt5` globally which is not consistent with MNE core

Reviewer 2: ```suggestion
The output file imports `PyQt5` globally, which is not consistent with MNE core
```

> structure so unit testing will fail. It is then strongly recommended to modify

Reviewer 2: ```suggestion
structure, causing unit testing to fail. It is then strongly recommended to modify
```

it as follows:

- Refactor the Qt version checking into a `_check_version` function

> - Use local import of `QtCore` in each function definition instead of globally

Reviewer 2: ```suggestion
- Use a local (instead of a global) import of `QtCore` in each function definition
```

- Do not call `qInitResources()` from the resource file itself, this function is
  called externally

Note
====

> The output file does not follow PEP8 guidelines so minor code formatting is

Reviewer 2: I guess there's something missing here? "minor code formatting deviations from PEP8" or something..?

.. -*- mode: rst -*-

Documentation
=============

The images are used in ``mne/viz/_brain`` as icons in the toolbar of ``_TimeViewer``.
It is necessary to compile those images into a resource file for proper use by
the application.

The resource configuration file ``mne.qrc`` describes the location of the resources
in the filesystem and also defines aliases for their use in the code.

To automatically generate the resource file in ``mne/viz/_brain``:

.. code-block:: bash

    pyrcc5 -o mne/viz/_brain/resources.py mne.qrc

Patching
========

> The output file imports ``PyQt5`` globally, which is not consistent with MNE core

Reviewer 3: Rather than all of this, can you nest the import of `resources.py` wherever it's actually used in the MNE code?

Author: I will try that right now

structure, causing unit testing to fail. It is then strongly recommended to modify
it as follows:

- Refactor the Qt version checking into a ``_check_version`` function
- Use a local (instead of a global) import of ``QtCore`` in each function definition
- Do not call ``qInitResources()`` from the resource file itself, this function is
  called externally

Note
====

> The output file contains minor code formatting deviations from PEP8:

Reviewer 3: ... that can be safely ignored, as the file is in the flake8 ignore list

(if it's not, please add it)
